src/lib/dataObjectUtils.ts

```diff
@@ -0,0 +1,422 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as dotWild from 'dot-wild';
+
+/**
+ * Checks if a value is an object.
+ *
+ * @param value - The value to be checked.
+ * @returns A boolean indicating whether the value is an object or not.
+ */
+const isObject = (value: any): boolean => (
```

Comment:
we have lodash, it has a function for this fyi

Reply:
I need it to have this implementation for the logic where its used